src/graphs/GraphAlgorithms.jl

```diff
@@ -116,4 +116,30 @@ function longest_paths(g::ACSet;
   lengths
 end
 
+"""Enumerate all paths of an acyclic graph, indexed by src+tgt"""
+function enumerate_paths(G::Graph;
+                         sorted::Union{AbstractVector{Int},Nothing}=nothing
+                        )::DefaultDict{Pair{Int,Int},Set{Vector{Int}}}
+  sorted = isnothing(sorted) ? topological_sort(G) : sorted
+  Path = Vector{Int}
+  paths = [Set{Path}() for _ in 1:nv(G)] # paths that start on a particular V
+  for v in reverse(topological_sort(G))
+    push!(paths[v], Int[]) # add length 0 paths
+    for e in incident(G, v, :src)
+      push!(paths[v], [e]) # add length 1 paths
+      for p in paths[G[e, :tgt]] # add length >1 paths
+        push!(paths[v], vcat([e], p))
+      end
+    end
+  end
+  # Restructure `paths` into a data structure indexed by start AND end V
+  allpaths = DefaultDict{Pair{Int,Int},Set{Path}}(()->Set{Path}())
```

Comment:
Not committed to this one, but I wonder if we should return an attributed reflexive graph, whose edges are paths in the original graphs with an attribute for the sequence forming the path?

Reply:
Yeah, I think that data structure better communicates what's going on! Unsure if creating a new attributed reflexive graph @theory is overkill for this, unless you can imagine us using that data structure in other places, too